### PR TITLE
Add priority for rendering

### PR DESCRIPTION
**Pull request by ivanPyrohivskyi**

### CODE REVIEW

File: native/src/rendering.cpp [hunk 1]
```diff
@@ -1027,6 +1028,22 @@ double polygonArea(MapDataObject* obj, float mult) {
 	return abs(area) * mult * mult * .5;
 }
 
+/*int getSquareSegmentLength(MapDataObject* obj) {
```
Comment (vshcherb) [inline native/src/rendering.cpp:1031]: no commented code, - please

File: native/src/rendering.cpp [hunk 2]
```diff
@@ -1082,6 +1099,13 @@ bool sortByOrder(const MapDataObjectPrimitive& i, const MapDataObjectPrimitive&
 	}
 	return (i.order < j.order);
 }
+bool sortByPriority(const MapDataObjectPrimitive& i, const MapDataObjectPrimitive& j) {
+	if (i.priority == j.priority) {
+		return sortByOrder(i, j);
+		//return getSquareSegmentLength(i.obj) < getSquareSegmentLength(j.obj);
```
Comment (vshcherb) [inline native/src/rendering.cpp:1105]: > return getSquareSegmentLength(i.obj) < getSquareSegmentLength(j.obj) - makes more sense as I see

File: native/src/rendering.cpp [hunk 3]
```diff
@@ -1027,6 +1028,22 @@ double polygonArea(MapDataObject* obj, float mult) {
...
 
+/*int getSquareSegmentLength(MapDataObject* obj) {
+	int dist = 0;
+	int prev_x = 0;
+	int prev_y = 0;
+	for (uint k = 0; k < obj->points.size(); k++) {
+		int x31 = obj->points[k].first;
+		int y31 = obj->points[k].second;
+		if (prev_x != 0 && prev_y != 0) {
+			dist += (int)(squareDist31TileMetric(prev_x, prev_y, x31, y31));
```
Comment (vshcherb) [inline native/src/rendering.cpp:1039]: easy overflow possible, why not use float?

File: native/src/rendering.cpp (hunk 2 above)
Comment (ivanPyrohivskyi) [reply native/src/rendering.cpp:1105]: getSquareSegmentLength:
 

sortByOrder:

File: native/src/rendering.cpp (hunk 1 above)
Comment (ivanPyrohivskyi) [reply native/src/rendering.cpp:1031]: It was specially left for review. See comment below.

File: native/src/renderRules.h [hunk 4]
```diff
@@ -472,6 +473,8 @@ class RenderingRulesStorageProperties
 		R_ONEWAY_ARROWS_COLOR = registerRuleInternal(RenderingRuleProperty::createOutputColorProperty("onewayArrowsColor"));
 		R_IGNORE_POLYGON_AS_POINT_AREA = registerRuleInternal(RenderingRuleProperty::createOutputBooleanProperty("ignorePolygonAsPointArea"));
 		R_ADD_POINT = registerRuleInternal(RenderingRuleProperty::createOutputBooleanProperty("addPoint"));
+
+		R_RENDERING_PRIORITY = registerRuleInternal(RenderingRuleProperty::createOutputIntProperty("renderingPriority"));
```
Comment (vshcherb) [inline native/src/renderRules.h:477]: naming is not good, it's not rendering priority. It's priority that's being used during density filter.

File: native/src/rendering.cpp [hunk 5]
```diff
@@ -1082,6 +1099,13 @@ bool sortByOrder(const MapDataObjectPrimitive& i, const MapDataObjectPrimitive&
 	}
 	return (i.order < j.order);
 }
+bool sortByPriority(const MapDataObjectPrimitive& i, const MapDataObjectPrimitive& j) {
```
Comment (vshcherb) [inline native/src/rendering.cpp:1102]: Also naming is not good not clear sort by which priority

**Pull request merged**